torch/csrc/jit/fuser/partition_desc.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "torch/csrc/WindowsTorchApiMacro.h"
+#include "torch/csrc/jit/assertions.h"
+#include "torch/csrc/jit/fuser/tensor_desc.h"
+
+#include <memory>
+#include <cstdint>
+#include <vector>
+
+namespace torch { namespace jit { namespace fuser {
+
+// Descriptor for chunk-ing an input tensor into subtensors
+// OR concat-ing an output tensor from subtensors
+struct TORCH_API PartitionDesc {
+  PartitionDesc()
```

Comment:
That's a fine way to think of them but the current code uses the default constructor for "pass-through" inputs and outputs that don't participate in concat/chunk operations. This can be seen in codegen.cpp, for example. I think we can simplify the shape inference more than is done in this PR but I don't want to extend it. The extended fusion logic requires a shape inference cleanup, anyway.